config: support relative path in --config (#9496)
tarantool/tarantool: changelogs/unreleased/config-relative-config-path.md

## bugfix/config

> * Support relative path in `--config <...>` option. Before this change
>   `config:reload()` fails in this case if `process.work_dir` option is set to a
>   non-null value (gh-8862).

Reviewer: ```suggestion
* Added support for the relative path in the `--config <...>` option. Before this change,
  `config:reload()` failed  if the `process.work_dir` option was set to a
  non-null value (gh-8862).
```

Author: Thanks! Updated.
